Description: fix errors with the TRUNCATE request
 The TRUNCATE SQL request is incompatible with MySQL an causes deadlock with PostgreSQL. Use DELETE instead.
Author: Nicolas Capit <nicolas.capit@imag.fr>
Origin: upstream
Bug: https://github.com/oar-team/oar/issues/11
Bug-Debian: #773790
Applied-Upstream: 3e2d9d88d3ae62f8308c848bcfde125f29f9c8d3
Index: oar/sources/core/common-libs/lib/OAR/IO.pm
===================================================================
--- oar.orig/sources/core/common-libs/lib/OAR/IO.pm
+++ oar/sources/core/common-libs/lib/OAR/IO.pm
@@ -6366,12 +6366,9 @@ sub update_gantt_visualization($){
 
     lock_table($dbh, ["gantt_jobs_predictions_visu","gantt_jobs_resources_visu","gantt_jobs_predictions","gantt_jobs_resources"]);
 
-#    $dbh->do("DELETE FROM gantt_jobs_predictions_visu");
-#    $dbh->do("DELETE FROM gantt_jobs_resources_visu");
+    $dbh->do("DELETE FROM gantt_jobs_predictions_visu");
+    $dbh->do("DELETE FROM gantt_jobs_resources_visu");
 ##    $dbh->do("OPTIMIZE TABLE ganttJobsResources_visu, ganttJobsPredictions_visu");
-    $dbh->do("TRUNCATE TABLE gantt_jobs_predictions_visu");
-    $dbh->do("TRUNCATE TABLE gantt_jobs_resources_visu");
-
     $dbh->do("INSERT INTO gantt_jobs_predictions_visu
               SELECT *
               FROM gantt_jobs_predictions
@@ -6482,8 +6479,8 @@ sub gantt_flush_tables($$$){
                         $sql
                  ");
     }else{
-        $dbh->do("TRUNCATE TABLE gantt_jobs_resources");
-        $dbh->do("TRUNCATE TABLE gantt_jobs_predictions");
+        $dbh->do("DELETE FROM gantt_jobs_predictions");
+        $dbh->do("DELETE FROM gantt_jobs_resources");
     }
 }
 
